src/core/homepage_elements/journals_and_html/hooks.py

```diff
@@ -0,0 +1,55 @@
+__copyright__ = "Copyright 2017 Birkbeck, University of London"
+__author__ = "Martin Paul Eve & Andy Byers"
+__license__ = "AGPL v3"
+__maintainer__ = "Birkbeck Centre for Technology and Publishing"
+
+import random
+
+from journal import models as journal_models
+from core.homepage_elements.journals_and_html import plugin_settings
+from utils.logger import get_logger
+from press import models as press_models
+
+logger = get_logger(__name__)
+
+
+def get_random_journals():
+    journals = journal_models.Journal.objects.filter(hide_from_press=False)
+    journal_pks = [journal.pk for journal in journals]
+    random_journal_pks = list()
+
+    random.shuffle(journal_pks)
+    for i in range(0, 6):
+
+        if journal_pks:
+            choice = journal_pks.pop()
+            random_journal_pks.append(choice)
```

Comment:
I think you are re-inventing `random.sample` here:
https://docs.python.org/3/library/random.html#random.sample

Reply:
random.sample does not work on querysets. It also doesn't work when the population is less than the sample.